### PR TITLE
Add `Popup.Color` Null Validation

### PR DESCRIPTION
**Pull request by brminnick**
### Description of Change ###

This PR ensures that `Popup.Color` cannot be set to `null`.

Because `Popup.Color` can be used as a bindable property, I implemented the `null` validation on `public static readonly BindableProperty ColorProperty` to ensure validation also happens if a `null` value is passed in via a binding.

 ### Additional information ###

I also added `ArgumentNullException.ThrowIfNull` to `WrapperControl` which is used by `Popup` on Windows.

### CODE REVIEW

**Comment by pictos** [comment]
With this we can remove some defensive to handle `null` values, like [this one](https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.windows.cs#L19). Do you mind if I add these changes on this PR?

**Comment by brminnick** [comment]
Good catch! Please do! 🙌

File: src/CommunityToolkit.Maui/Views/WrapperControl.windows.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿using Microsoft.Maui.Platform;
+﻿using System;
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Views/WrapperControl.windows.cs:1]:
```suggestion
```

**Review by VladislavAntonyuk — APPROVED** [review]

**Review by VladislavAntonyuk — APPROVED** [review]

**Review by pictos — APPROVED** [review]
We can't remove the null checks, because on `IPopup` we set the Color as `Color?`, so other implementations can allow null if they want. The only way to remove the defensive code is by changing `Color?` to `Color`.

**Pull request merged**